docs/db2graph/db2graph.rst

```diff
@@ -0,0 +1,253 @@
+           relation_name_B_to_C -- this is the name of the edge from B to C
+           SELECT table1_name.column_name_B, table2_name.column_name_C FROM table1_name, table2_name WHERE ...; -- this row represents an edge from source entity node B to destination entity node C
+
+   | The user can expand or shorten the list of queries in the above query definition file to query a certain subset of data from the database.
+
+   .. note:: 
+       Db2Graph validates the correctness of format of each query. However, it does not validate the correctness of the queries. That is, it assumes that all column names and table names exist in the given database schema provided by the user. An error will be thrown in the event that the validation check fails.
+    
+#. | Lastly, execute Db2Graph with the following commands. Two flags are required. Note that prints will include both errors and general information, and those are also logged to ``./output_dir/output.log``:
+
+    .. code-block:: bash
+        
+           $ MARIUS_NO_BINDINGS=1 marius_db2graph --config_path conf/config.yaml --output_directory output_dir/
+           Starting a new run!!!
+           ...
+           Edge file written to output_dir/edges.txt
+
+   | The  ``--config_path`` flag specifies where the configuration file created by the user is.
+
+   | The  ``--output_directory`` flag specifies where the data will be output and is set by the user. In this example, assume we have not created the output_dir directory. ``db2graph`` will create it for us. 
+
+   | The conversion result will be written to ``edges.txt`` in a newly created directory named ``./output_dir``:
+    
+    .. code-block:: bash
+        
+           $ ls -l .
+           output_dir/
+             edges.txt                       # generated file with sets of triples
+             output.log                          # output log file
+           conf/  
+             config.yaml                         # config file
+             edges_queries.txt             # defines edges_queries    
+          $ cat output_dir/edges.txt
+          column_name_A    relation_name_A_to_B    column_name_B
+          column_name_B    relation_name_B_to_C    column_name_C
+    
+Example Use Case
+""""""""""""""""""""
+
+We use `The Movie Dataset <https://www.kaggle.com/datasets/rounakbanik/the-movies-dataset>`_ from Kaggle as an example to demonstrate a step-by-step walkthrough from loading a dataset into a PostgreSQL database to defining the edge queries and to converting the database into a graph using Db2Graph. Note the following steps assume the database has not been created and Marius has not been installed.
+
+#. | First, create a docker container from the PostgreSQL image. This container will contain all of our work in this example. Note that the password of this container, ``password``, will be the password of the database we create.
```

Comment:
Is it necessary for this example to include a docker container? It adds another level of complexity.

If it is necessary then we should create a dockerfile which contains the necessary setup, package installs, and data downloads. I tried to run this example but had to resolve a ton of package not installed issues (had to update apt, install vim, git, unzip, wget, python3, pip3, pandas, psycopg2)